src/qmflows/parsers/cp2KParser.py

```diff
@@ -90,28 +93,70 @@ def read_cp2k_coefficients(path_mos: PathLike,
+        printed_orbitals = range_mos[1] - range_mos[0] + 1
+
+        return read_coefficients(path_mos, printed_orbitals, orbitals_info.nOrbFuns)
+    except ValueError as err:
+        msg = f"""There was a problem reading the molecular orbitals from:{path_mos}\n
+contact the developers!!\nValueError: {err}"""
+        logger.error(msg)
+        raise
+    except TypeError:
+        msg = f"""There was a problem reading the ``range_mos` parameter.
+Its value is {range_mos}"""
+        logger.error(msg)
+        raise
+
+
+def read_coefficients(path: PathLike, nOrbitals: int, nOrbFuns: int) -> InfoMO:
+    """Read the coefficients from the plain text output.
 
-    return readCp2KCoeff(path_mos, printed_orbitals, orbitals_info.nOrbFuns)
+    Notes
+    -----
+    MO coefficients are stored in Column-major order.
+    CP2K molecular orbitals output looks like:
 
+    MO EIGENVALUES, MO OCCUPATION NUMBERS, AND SPHERICAL MO EIGENVECTORS
 
-def readCp2KCoeff(path: PathLike, nOrbitals: int, nOrbFuns: int) -> InfoMO:
-    """Read the coefficients from the plain text output.
+                              5                      6
+                          -0.2590267204166110    -0.1785544120250688
 
-    MO coefficients are stored in Column-major order.
+                           2.0000000000000000     2.0000000000000000
 
-    :parameter path: Path to the file containing the MO coefficients
-    :type path: String
-    :parameter nOrbitals: Number of MO to read
-    :param nOrbFuns: Number of orbital functions
-    :returns: Molecular orbitals and orbital energies
+     1     1  C  2s        0.0021482361354044     0.0000000235522485
+     2     1  C  3s       -0.0007100065367389     0.0000000102096730
+     3     1  C  3py      -0.1899052318987045     0.0000000059435027
+     4     1  C  3pz       0.0000000178537720    -0.5500605729231620
+     5     1  C  3px       0.3686765614894165     0.0000000228716009
+     6     1  C  4py       0.0014072130025389     0.0000000019199413
+     7     1  C  4pz      -0.0000000014121887     0.0293850516018881
+     8     1  C  4px      -0.0028383911872079     0.0000000042372601
+     9     1  C  4d-2      0.0311183981707317     0.0000000014108937
+    10     1  C  4d-1      0.0000000095952723     0.0253837978837068
+    11     1  C  4d0       0.0005419630026689     0.0000000391888080
+    12     1  C  4d+1     -0.0000000210955114     0.0147105486663415
+    13     1  C  4d+2      0.0534202997324328     0.0000000021056315
+
+    Parameters
+    ----------
+    path
+        Path to the file containing the MO coefficients
+    nOrbitals
+        Number of MO to read
+    nOrbFuns
+        Number of orbital functions
+
+    Returns
+    -------
+        Molecular orbitals and orbital energies
     """
     def remove_trailing(xs):
         """Remove the last lines of the MOs output."""
-        words = ['Fermi', 'HOMO-LUMO']
+        words = ('Fermi', 'HOMO-LUMO')
         if any([x in words for x in xs[-1]]):
```

Comment:
```suggestion
        if any(x in words for x in xs[-1]):
```

Reply:
This can actually be faster than list comprehension, as iteration through the generator will terminate the moment `x in words` evaluates to `True`.

Reply:
Thanks, I am including them :+1: